### PR TITLE
chore(Dialog): Dialog wrapper uses `data-iui-relative` instead of inline positioning

### PR DESCRIPTION
**Pull request by r100-stack**
* Implements [v1 (issue #831)](https://github.com/iTwin/iTwinUI-react/issues/831) CSS change: [fix: Make dialog positioned and remove unneeded styles](https://github.com/iTwin/iTwinUI/pull/756/commits/da810533f758fae1a0e8a61ad4144a88519be178)
* Dialog wrapper sets `data-iui-relative={relativeTo === 'container'}` instead of inline style `position: relativeTo === 'container' ? 'absolute' : 'fixed'`
* Removes double prop deconstruction in `DialogBackdrop.tsx` that happened due to PR [#866](https://github.com/iTwin/iTwinUI-react/pull/866) as mentioned in [#866 (comment)](https://github.com/iTwin/iTwinUI-react/pull/866#discussion_r993542560)

## Checklist

- [x] Add meaningful unit tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
  - **IGNORING** `Select -> Truncate Middle Text` since it even fails before this PR.
- ~[ ]~ Add component features demo in Storybook (different stories)
- ~[ ]~ Approve test images for new stories
- ~[ ]~ Add screenshots of the key elements of the component

### CODE REVIEW

**Review by elephantcatdog — APPROVED** [review]
Looks good to me

**Comment by gretanausedaite** [comment]
I think new CSS broke resizing. We need to add back `max-width: 100%` to `iui-dialog-draggable`.
Current css:
https://user-images.githubusercontent.com/83585998/195830957-5fb54c95-48dd-4d4a-89da-9910eb4c86b9.mp4

Added back max-width:
https://user-images.githubusercontent.com/83585998/195831043-443ed215-a752-4947-9e84-373c87c470a0.mp4

**Comment by rohan-kadkol** [comment]
> I think new CSS broke resizing. We need to add back `max-width: 100%` to `iui-dialog-draggable`. Current css:
> 
>  Screen.Recording.2022-10-14.at.13.53.26.mp4 
> Added back max-width:
> 
>  Screen.Recording.2022-10-14.at.13.54.25.mp4

Good catch! Just for context, I think [L113-116](https://github.com/iTwin/iTwinUI/pull/773/files#diff-68df02c08d0adce541e49bba164c2c72d0c346f659f79c8a76495a884769741aL113-L116) are the changes you are referring to. 

Adding `max-width` in the CSS repo back doesn't make any visual change, but since we can't resize dialogs in the CSS repo, it may be unnecessary to add there. So I added an inline style on [L203](https://github.com/iTwin/iTwinUI-react/pull/873/files#diff-18990ed4610236c6006f67085d05378e572b80f148f75261524c2f8f3fc103f3R203).

File: packages/iTwinUI-react/src/core/utils/components/Resizer.tsx [hunk 1]
```diff
@@ -74,6 +81,15 @@ export const Resizer = (props: ResizerProps) => {
         return;
       }
 
+      if (!isResizing.current) {
+        isResizing.current = true;
+        onResizeStart?.({
+          width,
+          height,
+          transform: `translate(${translateX}px, ${translateY}px)`,
```
Comment (mayank99) [inline packages/iTwinUI-react/src/core/utils/components/Resizer.tsx:89]: these arguments are not used anywhere so i'd remove them

File: packages/iTwinUI-react/src/core/Dialog/DialogMain.tsx [hunk 2]
```diff
@@ -210,6 +209,7 @@ export const DialogMain = React.forwardRef<HTMLDivElement, DialogMainProps>(
           <Resizer
             elementRef={dialogRef}
             containerRef={dialogContext.dialogRootRef}
+            onResizeStart={() => setResizeStyle({ maxWidth: '100%' })}
```
Comment (mayank99) [inline packages/iTwinUI-react/src/core/Dialog/DialogMain.tsx:212]: This only needs to happen once so i would add guard this in an if condition (can use a boolean ref)

Comment (rohan-kadkol) [reply packages/iTwinUI-react/src/core/Dialog/DialogMain.tsx]: Thanks, added a guard condition using a boolean ref

Comment (mayank99) [reply packages/iTwinUI-react/src/core/Dialog/DialogMain.tsx]: Please remove the console log

Comment (rohan-kadkol) [reply packages/iTwinUI-react/src/core/Dialog/DialogMain.tsx]: [DONE] Oops! :smile: Removed

**Review by mayank99 — APPROVED** [review]

**Comment by rohan-kadkol** [comment]
> > I think new CSS broke resizing. We need to add back `max-width: 100%` to `iui-dialog-draggable`. Current css:
> > Screen.Recording.2022-10-14.at.13.53.26.mp4
> > Added back max-width:
> > Screen.Recording.2022-10-14.at.13.54.25.mp4
> 
> Good catch! Just for context, I think [L113-116](https://github.com/iTwin/iTwinUI/pull/773/files#diff-68df02c08d0adce541e49bba164c2c72d0c346f659f79c8a76495a884769741aL113-L116) are the changes you are referring to.
> 
> Adding `max-width` in the CSS repo back doesn't make any visual change, but since we can't resize dialogs in the CSS repo, it may be unnecessary to add there. So I added an inline style on [L203](https://github.com/iTwin/iTwinUI-react/pull/873/files#diff-18990ed4610236c6006f67085d05378e572b80f148f75261524c2f8f3fc103f3R203).

Just clarifying the latest changes before merging. As discussed previously, `.iui-dialog-draggable` starts with the CSS style `max-width: max(50%, 380px)` and when it is resized for the first time, it gets the inline style `max-width: 100%`. ([Commits](https://github.com/iTwin/iTwinUI-react/pull/873/files/36452a75d51d82156142d7c5e2313b4b64c76478..c9d5243d380e09da909ae3585244e5957c3e2064))

**Review by gretanausedaite — APPROVED** [review]

**Pull request merged**